tools/modules/framework/module_data_files.sh

```
@@ -0,0 +1,391 @@
+
+		# Call the specified generator function
+		if [[ $(type -t "$generator") == "function" ]]; then
+			"$generator"
+		else
+			echo "Error: Invalid generator function '$generator'."
+			return 1
+		fi
+
+	done
+
+	chown -R "${SUDO_USER:-$USER}":"${SUDO_USER:-$USER}" "$tools_dir"
```

Shellcheck (0.10.0)

[warning] tools_dir is referenced but not assigned.

(SC2154)